qiskit/transpiler/passes/layout/vf2_layout.py

```diff
@@ -80,14 +81,26 @@ def __init__(
                 based on the number of edges in the interaction graph or the coupling graph
                 (whichever is larger). If set to a value <= 0 no limit on the number of trials
                 will be set.
+            target (Target): A target representing the backend device to run vf2layout on.
+                If specified it will supersede a set value for ``properties`` and
+                ``coupling_map``.
+
+        Raises:
+            TypeError: If neither ``coupling_map`` or ``target`` are provided.
         """
         super().__init__()
-        self.coupling_map = coupling_map
+        self.target = target
+        if target is not None:
+            self.coupling_map = self.target.build_coupling_map()
+        elif coupling_map is not None:
+            self.coupling_map = coupling_map
+        else:
+            raise TypeError("coupling_map or target must be specified.")
```

Comment:
Hah, fun I forgot that in the preset pass managers a coupling map isn't required to be set (making this pass a no-op), which this breaks

Reply:
Fixed in: https://github.com/Qiskit/qiskit-terra/pull/7735/commits/23c0ad1451e416709da434df965168a3609a73f6